include/libopencm3/nrf/common/radio.h

```diff
@@ -0,0 +1,325 @@
+
+#define RADIO_PCNF0_S1LEN_SHIFT			(16)
+#define RADIO_PCNF0_S1LEN_MASK			(0xf << RADIO_PCNF0_S1LEN_SHIFT)
+#define RADIO_PCNF0_S1LEN_MASKED(V)			(((V) << RADIO_PCNF0_S1LEN_SHIFT) & \
+		RADIO_PCNF0_S1LEN_MASK)
+
+#define RADIO_PCNF1_MAXLEN_SHIFT			(0)
+#define RADIO_PCNF1_MAXLEN_MASK			(0xff << RADIO_PCNF1_MAXLEN_SHIFT)
+#define RADIO_PCNF1_MAXLEN_MASKED(V)			(((V) << RADIO_PCNF1_MAXLEN_SHIFT) & \
+		RADIO_PCNF1_MAXLEN_MASK)
+
+#define RADIO_PCNF1_STATLEN_SHIFT			(8)
+#define RADIO_PCNF1_STATLEN_MASK			(0xff << RADIO_PCNF1_STATLEN_SHIFT)
+#define RADIO_PCNF1_STATLEN_MASKED(V)			(((V) << RADIO_PCNF1_STATLEN_SHIFT) & \
+		RADIO_PCNF1_STATLEN_MASK)
+
+#define RADIO_PCNF1_BALEN_SHIFT			(16)
+#define RADIO_PCNF1_BALEN_MASK			(7 << RADIO_PCNF1_BALEN_SHIFT)
+#define RADIO_PCNF1_BALEN_MASKED(V)			(((V) << RADIO_PCNF1_BALEN_SHIFT) & \
+		RADIO_PCNF1_BALEN_MASK)
+
+#define RADIO_PCNF1_ENDIAN_BIG			(1 << 24)
+#define RADIO_PCNF1_WHITEEN			(1 << 25)
+
+#define RADIO_PREFIX0_AP0_SHIFT			(0)
+#define RADIO_PREFIX0_AP0_MASK			(0xff << RADIO_PREFIX0_AP0_SHIFT)
+#define RADIO_PREFIX0_AP0_MASKED(V)			(((V) << RADIO_PREFIX0_AP0_SHIFT) & \
+		RADIO_PREFIX0_AP0_MASK)
+
+#define RADIO_PREFIX0_AP1_SHIFT			(8)
+#define RADIO_PREFIX0_AP1_MASK			(0xff << RADIO_PREFIX0_AP1_SHIFT)
+#define RADIO_PREFIX0_AP1_MASKED(V)			(((V) << RADIO_PREFIX0_AP1_SHIFT) & \
+		RADIO_PREFIX0_AP1_MASK)
+
+#define RADIO_PREFIX0_AP2_SHIFT			(16)
+#define RADIO_PREFIX0_AP2_MASK			(0xff << RADIO_PREFIX0_AP2_SHIFT)
+#define RADIO_PREFIX0_AP2_MASKED(V)			(((V) << RADIO_PREFIX0_AP2_SHIFT) & \
+		RADIO_PREFIX0_AP2_MASK)
+
+#define RADIO_PREFIX0_AP3_SHIFT			(24)
+#define RADIO_PREFIX0_AP3_MASK			(0xff << RADIO_PREFIX0_AP3_SHIFT)
+#define RADIO_PREFIX0_AP3_MASKED(V)			(((V) << RADIO_PREFIX0_AP3_SHIFT) & \
+		RADIO_PREFIX0_AP3_MASK)
+
+#define RADIO_PREFIX1_AP4_SHIFT			(0)
+#define RADIO_PREFIX1_AP4_MASK			(0xff << RADIO_PREFIX1_AP4_SHIFT)
+#define RADIO_PREFIX1_AP4_MASKED(V)			(((V) << RADIO_PREFIX1_AP4_SHIFT) & \
+		RADIO_PREFIX1_AP4_MASK)
+
+#define RADIO_PREFIX1_AP5_SHIFT			(8)
+#define RADIO_PREFIX1_AP5_MASK			(0xff << RADIO_PREFIX1_AP5_SHIFT)
+#define RADIO_PREFIX1_AP5_MASKED(V)			(((V) << RADIO_PREFIX1_AP5_SHIFT) & \
+		RADIO_PREFIX1_AP5_MASK)
+
+#define RADIO_PREFIX1_AP6_SHIFT			(16)
+#define RADIO_PREFIX1_AP6_MASK			(0xff << RADIO_PREFIX1_AP6_SHIFT)
+#define RADIO_PREFIX1_AP6_MASKED(V)			(((V) << RADIO_PREFIX1_AP6_SHIFT) & \
+		RADIO_PREFIX1_AP6_MASK)
+
+#define RADIO_PREFIX1_AP7_SHIFT			(24)
+#define RADIO_PREFIX1_AP7_MASK			(0xff << RADIO_PREFIX1_AP7_SHIFT)
+#define RADIO_PREFIX1_AP7_MASKED(V)			(((V) << RADIO_PREFIX1_AP7_SHIFT) & \
+		RADIO_PREFIX1_AP7_MASK)
+
+#define RADIO_PREFIX_AP(n)			((n) < 4 ? RADIO_PREFIX0 : RADIO_PREFIX1)
+#define RADIO_PREFIX_AP_SHIFT(n)			(8 * (n & 3))
+#define RADIO_PREFIX_AP_MASK(n)			(0xff << RADIO_PREFIX_AP_SHIFT(n))
+#define RADIO_PREFIX_AP_MASKED(n, V)			(((V) << RADIO_PREFIX_AP_SHIFT(n)) & \
+		RADIO_PREFIX_AP_MASK(n))
+
+/* TODO: Get rid of this */
```

Comment:
ok, let's do that?

Reply:
I assume that code from Unicore works. As I don't have working implementation of radio yet I'd rather drag this in. Ugly, but working. And then clean it up once I can guarrantee that it will still be working.

Reply:
hrm, ok. big assumption, but we can go with that :)